src/langchain_google_firestore/vectorstores.py

```diff
@@ -341,5 +434,46 @@ def from_texts(
         vs_obj.add_texts(texts, metadatas, ids, **kwargs)
         return vs_obj
 
+    def _encode_image(self, uri: str) -> str:
+        """Get base64 string from a image URI."""
+        gcs_uri = re.match("gs://(.*?)/(.*)", uri)
+        if gcs_uri:
+            bucket_name, object_name = gcs_uri.groups()
+            storage_client = storage.Client()
+            bucket = storage_client.bucket(bucket_name)
+            blob = bucket.blob(object_name)
+            return base64.b64encode(blob.download_as_bytes()).decode("utf-8")
+
+        web_uri = re.match(r"^(https?://).*", uri)
+        if web_uri:
+            response = requests.get(uri, stream=True)
+            response.raise_for_status()
+            return base64.b64encode(response.content).decode("utf-8")
+
+        with open(uri, "rb") as image_file:
+            return base64.b64encode(image_file.read()).decode("utf-8")
+
+    def _images_embedding_helper(self, image_uris: List[str]) -> List[List[float]]:
+        if not hasattr(self.embedding_service, "embed_image"):
```

Comment:
The `embed_image` API will be replaced by `embed_images` for VertexAIEmbeddings: https://github.com/langchain-ai/langchain-google/pull/495 so we need to check for both attributes

Reply:
Thanks @duwenxin99 for the note! I changed the code to handle both `embed_image` and `embed_images`. I also have an extra check in `embed_image` to handle both single images and a list of images because `VertexAIEmbedding` currently only has a single image.